examples/new_theme.py

```diff
@@ -0,0 +1,27 @@
+"""
+Displays an image and sets the theme to new custom theme.
+"""
+
+from skimage import data
+import napari
+from napari.utils.theme import palettes
```

Comment:
To me, it's still unclear why these are called palettes and not themes. To me a palette is a set of colours — not an assignment of elements to those colours. Looking it up in the dictionary, we have "the range of colours used by a particular artist or in a particular picture", or "(computer graphics) the range of colours or shapes available to the user". So I really think it's the wrong word here.

Additionally, either through magic or through changing this to a function, it would be good if this *already* returned a copy of the theme. e.g.

```python
from napari.utils.theme import get_themes, register_theme


blue_theme = {
    **get_themes('dark'),
    'background': 'rgb(28, 31, 48)',
    'foreground': 'rgb(45, 52, 71)',
    'primary': 'rgb(80, 88, 108)',
    'current': 'rgb(184, 112, 0)',
}

register_theme('blue', blue_theme)
```

If you don't like the `{**}` notation, you can also do it more conventionally:

```python
from napari.utils.theme import get_themes, register_theme


blue_theme = get_themes('dark')
blue_theme.update(
    background='rgb(28, 31, 48)',
    foreground='rgb(45, 52, 71)',
    primary='rgb(80, 88, 108)',
    current='rgb(184, 112, 0)',
)

register_theme('blue', blue_theme)
```

But the key is that the user is never interacting with a global dictionary, and we don't rely on them to remember to `.copy()`. The only ever receive their own copies.

Reply:
Ok, all the requested changes have been made. I feel like they could have come in a follow up PR as this PR didn't directly touch those methods, but I see how it's nice to do it all it at once and it wasn't too much effort.

I would request though that if you have further requested changes (like different names etc) you can make them yourself and push them directly to this PR.

I've also added tests for the new methods